invenio_rdm_records/resources/resources.py

```diff
@@ -76,10 +76,89 @@ def p(route):
                 p(routes["access-request-settings"]),
                 self.update_access_settings,
             ),
+            route("POST", p(routes["delete-record"]), self.delete_record),
+            route("POST", p(routes["restore-record"]), self.restore_record),
+            route("GET", p(routes["list-all"]), self.search_all),
+            route("POST", p(routes["set-record-quota"]), self.set_record_quota),
+            # TODO: move to users?
+            route("POST", routes["set-user-quota"], self.set_user_quota),
         ]
 
         return url_rules
 
+    @request_headers
+    @request_view_args
+    @request_data
+    def set_record_quota(self):
+        """Read the related review request."""
+        item = self.service.set_quota(
+            g.identity,
+            resource_requestctx.view_args["pid_value"],
+            quota_size=resource_requestctx.data["quota_size"],
+            max_file_size=resource_requestctx.data["max_file_size"],
+            notes=resource_requestctx.data["notes"],
+        )
+
+        return {}, 200
+
+    @request_headers
+    @request_view_args
+    @request_data
+    def set_user_quota(self):
+        """Read the related review request."""
+        item = self.service.set_user_quota(
+            g.identity,
+            id_=resource_requestctx.view_args["pid_value"],
+            quota_size=resource_requestctx.data["quota_size"],
+            max_file_size=resource_requestctx.data["max_file_size"],
```

Comment:
this should go through marshmallow schema, but I didn't add it, I was not sure if it was done on purpose in the service this way

Reply:
+1 on adding param validation